src/extremeweatherbench/utils.py

```diff
@@ -403,3 +320,176 @@ def read_event_yaml(input_pth: str | Path) -> dict:
     with open(input_pth, "rb") as f:
         yaml_event_case = yaml.safe_load(f)
     return yaml_event_case
+
+
+def location_subset_point_obs(
+    df: pd.DataFrame,
+    min_lat: float,
+    max_lat: float,
+    min_lon: float,
+    max_lon: float,
+):
+    """Subset a dataframe based upon maximum and minimum latitudes and longitudes.
+
+    Arguments:
+        df: dataframe with point obs.
+        min_lat: minimum latitude.
+        max_lat: maximum latitude.
+        min_lon: minimum longitude.
+        max_lon: maximum longitude.
+
+    Returns a subset dataframe."""
+    location_subset_df = df[
+        (df["latitude"] >= min_lat)
+        & (df["latitude"] <= max_lat)
+        & (df["longitude"] >= min_lon)
+        & (df["longitude"] <= max_lon)
+    ]
+    return location_subset_df
+
+
+def align_point_obs_from_gridded(
+    forecast_ds: xr.Dataset,
+    case_subset_point_obs_df: pd.DataFrame,
+    data_var: List[str],
+    point_obs_metadata_vars: List[str],
+) -> Tuple[xr.Dataset, xr.Dataset]:
+    """Takes in a forecast dataarray and point observation dataframe, aligning them by
+    reducing dimensions.
+
+    Args:
+        forecast_ds: The forecast dataset.
+        case_subset_point_obs_df: The point observation dataframe.
+        data_var: The variable to subset (e.g. "surface_air_temperature")
+        point_obs_metadata_vars: The metadata variables to subset (e.g. ["elev", "name"])
+
+    Returns a tuple of aligned forecast and observation dataarrays."""
+    case_subset_point_obs_df = case_subset_point_obs_df[
+        POINT_OBS_METADATA_VARS + data_var
+    ]
+    case_subset_point_obs_df = case_subset_point_obs_df.rename(columns=ISD_MAPPING)
+    case_subset_point_obs_df["longitude"] = convert_longitude_to_360(
+        case_subset_point_obs_df["longitude"]
+    )
+    case_subset_point_obs_df = location_subset_point_obs(
+        case_subset_point_obs_df,
+        forecast_ds["latitude"].min().values,
+        forecast_ds["latitude"].max().values,
+        forecast_ds["longitude"].min().values,
+        forecast_ds["longitude"].max().values,
+    )
+    # Uses indexing in the dataframe to capture metadata columns for future use
+    point_obs_metadata = case_subset_point_obs_df[point_obs_metadata_vars]
+
+    # Reset index to allow for easier modification
+    case_subset_point_obs_df = case_subset_point_obs_df.reset_index()
+    case_subset_point_obs_df.loc[:, "station"] = case_subset_point_obs_df[
+        "station"
+    ].astype("str")
+
+    # Set up multiindex to enable slicing along individual timesteps
+    case_subset_point_obs_df = case_subset_point_obs_df.set_index(
+        ["station", "time"]
+    ).sort_index()
```

Comment:
This line is a little strange... do you mean to simply create a column named "station" with the values set to the original station but transformed to the `str` type?

If so, I would just do this in one fell swoop with a method chained block:

    
```suggestion 
    case_subset_point_obs_df = (
        case_subset_point_obs_df
        .reset_index()  # Reset index to allow for easier modification
        .assign(station=case_subset_point_obs_df["station"].astype("str")
        .set_index(["station", "time"])  # Set up multiindex to enable slicing along individual timesteps
        .sort_index()
     )
```